diff -up aria2-1.16.1/src/Makefile.in.build aria2-1.16.1/src/Makefile.in
--- aria2-1.16.1/src/Makefile.in.build	2012-12-26 17:09:00.000000000 +0100
+++ aria2-1.16.1/src/Makefile.in	2013-03-06 14:51:51.635190936 +0100
@@ -1422,7 +1422,7 @@ SRCS = SocketCore.cc SocketCore.h Binary
 	$(am__append_34) $(am__append_35)
 noinst_LIBRARIES = libaria2c.a
 libaria2c_a_SOURCES = $(SRCS)
-aria2c_LDADD = libaria2c.a @LIBINTL@ @ALLOCA@ #-lprofiler
+aria2c_LDADD = libaria2c.a @LIBINTL@ @ALLOCA@ -lpthread #-lprofiler
 #aria2c_LDFLAGS = -pg
 AM_CPPFLAGS = -Wall\
 	-I$(top_srcdir)/lib -I$(top_srcdir)/intl\
